packages/babel-parser/src/parser/expression.js

```diff
@@ -1214,6 +1214,26 @@ export default class ExpressionParser extends LValParser {
         return node;
       }
 
+      case tt.moduloAssign:
+        if (
+          this.getPluginOption("pipelineOperator", "proposal") === "hack" &&
+          this.getPluginOption("pipelineOperator", "topicToken") === "%"
+        ) {
+          // If we find %= in an expression position, and the Hack-pipes proposal is active,
+          // then the % could be the topic token (e.g., in x |> %==y or x |> %===y), and so we
+          // reparse it as %.
+          // The next readToken() call will start parsing from =.
+
+          this.state.value = "%";
+          this.state.type = tt.modulo;
+          this.state.pos--;
```

Comment:
`this.state.pos` should be `this.state.start + 1`, because a topic reference can be an LHS so whitespaces may follow `=`.

```js
function assign(n) {
  return n |> (
    %=
  2)
}
```